renovate.json

```diff
@@ -14,10 +14,10 @@
         "^rockcraft_rock/rockcraft.yaml$"
       ],
       "matchStrings": [
-        "source-tag: (?<currentValue>.*?)\\s"
+        "rockcraft$\s+source-tag:\s+(?<currentValue>.*?)"
```

Comment:
Does this work? I tried in [regex101.com](https://regex101.com/), but couldn't get it to match properly.

How about this?

```
rockcraft:$[\S\s]*?source-tag:\s+(?<currentValue>.*?)\s
```

Note that you will need to add backslashes before the backslashes, because they are inside double-quotes.
```suggestion
        "rockcraft:$[\\S\\s]*?source-tag:\\s+(?<currentValue>.*?)\\s"
```

Reply:
Y indeed was missing the escaped backslashes.

And y, it was working for me on regex101:
![image](https://github.com/canonical/rocks-toolbox/assets/4047767/f182785e-e7e3-4ed2-8c9b-332aa1eb4b00)

Yours isn't working for me though. What's your test string?

Reply:
Note that, there are other lines between `rockcraft:` and the `source-tag:` lines.

https://github.com/canonical/rocks-toolbox/blob/4ac589d9e70181e5a5f9e60203c46a6f4b4383f0/rockcraft_rock/rockcraft.yaml#L33-L36

![image](https://github.com/canonical/rocks-toolbox/assets/18555205/3c7bf5e8-54d9-4a4f-8e37-6ad48e242b6e)

Reply:
same:
![image](https://github.com/canonical/rocks-toolbox/assets/4047767/212434a4-aa4f-44ab-9fef-10d4a6e03255)

Are you using ECMAScript?

Reply:
Right, there was an extra white space after `rockcraft:`. This is too error-prone. Trying a different flavour

Reply:
What about `rockcraft\.git[\s]*?$[\s]*?source-tag:\s+(?<currentValue>.*?)\s`? 
![image](https://github.com/canonical/rocks-toolbox/assets/4047767/ecf9c6ad-bc2e-4da8-9bed-b8271859012e)

![image](https://github.com/canonical/rocks-toolbox/assets/4047767/5ecc10bb-4d73-4e11-acf2-9d54b62ffbc5)
![image](https://github.com/canonical/rocks-toolbox/assets/4047767/a6d26833-c4fb-44f8-8d5a-7a3c0f30741b)

Reply:
Sounds good!

Reply:
done https://github.com/canonical/rocks-toolbox/pull/15/commits/f19b7c5104c38611807aec53495625c4755265e4